src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AgreementsApiServiceImpl.scala

```diff
@@ -486,4 +486,25 @@ final case class AgreementsApiServiceImpl(
     }
   }
 
+  override def getAgreementProducers(q: Option[String], offset: Int, limit: Int)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerCompactOrganizations: ToEntityMarshaller[CompactOrganizations]
+  ): Route = {
+    val result: Future[CompactOrganizations] =
+      for {
+        pagedResults <- agreementProcessService.getAgreementProducers(q, offset = offset, limit = limit)
```

Comment:
I was missing one requirement: `q` length should be ≥ 3
So cases are:
- `q` is `None` -> return all producers
- `q` is `Some` and length = 0 -> return all producers
- `q` is `Some` and length > 0 and < 3 -> return and empty list
- `q` is `Some` and length ≥ 3 -> return the filtered list

Reply:
il terzo caso forse sarebbe meglio rendere un eccezione 400?